Add docs for infrastructure hosts feature (#2756)
elastic/observability-docs: docs/en/observability/monitor-infra/analyze-hosts.asciidoc

then click **Hosts**. If the view is not enabled, click **Enable host view** to
begin exploring.

include::view-infrastructure-metrics.asciidoc[tag=add-metrics-tip]

[role="screenshot"]
image::images/hosts.png[Screenshot of the Hosts page]

The **Hosts** page provides a few different ways to view host metrics:

* Overview tiles show the number of hosts returned by your search plus
averages of key metrics, including CPU usage, memory usage, and throughput.
* The host list shows all the hosts returned by your search with a
breakdown of metrics for each host. You may need to page through the list or
change the number of rows displayed on each page to see all of your hosts.
* Each host name is an active link to an overview page with additional metrics
about the host, such as CPU usage, load, memory usage, and network traffic.
* The **Metrics** view shows metrics trending over time for the top 20 hosts,
including normalized load, CPU usage, memory usage, network inbound, network
outbound, disk read IOPS, and disk write IOPS. Place your cursor over a line to
view metrics at a specific point in time.

[NOTE]
====

> This feature is in technical preview. We'd love to get your feedback!

Reviewer: 👍 Done in 40b6ba4c1b4c613363b7e8b4b9106093c9c23cd3.

> This feature is in technical preview. We'd love to get your feedback!

Reviewer 2: Thanks!

https://docs.google.com/forms/d/e/1FAIpQLScRHG8TIVb1Oq8ZhD4aks3P1TmgiM58TY123QpDCcBz83YC6w/viewform[Tell us what you think!]
====

[discrete]
[[add-data-hosts-page]]
== Add data

The **Hosts** page displays metrics from indices that match the index patterns
specified on the **Settings** page. For example, the default patterns are 
`metrics-*` and `metricbeat-*`. 

To add metrics:

. Go to the **Hosts** page.
. If there are no metrics to display, {kib} will prompt you to add an
integration. Otherwise click **Add data**.
. Search for an integration that matches the type of host you want to monitor
and add the integration.
. Follow the in-product instructions to install the integration plus any
required programs, like {agent} or {metricbeat}.

To learn more about adding observability data, refer to
<<add-observability-data>>.

[discrete]
[[analyze-hosts-filter-view]]
== Filter the Hosts view

The **Hosts** page provides several mechanisms for filtering the data on the
page:

* Enter a search query to show metrics that match your search criteria. For
example, to see metrics for all hosts running on Google Cloud Platform (GCP),

> enter `cloud.account.provider : "gcp"`. Otherwise you'll see metrics for all

Author: @roshan-elastic I need to replace this with a useful example and maybe add a couple more examples to this section. Any ideas?

Reviewer: 👍 Changed in 40b6ba4c1b4c613363b7e8b4b9106093c9c23cd3.

open the visualization in Lens.

In Lens, you can examine all the fields and formulas used to create the
visualization, make modifications to the visualization, and save your changes.

For more information about using Lens, refer to the
{kibana-ref}/lens.html[{kib} documentation about Lens].

[discrete]
[[analyze-hosts-inspect-data]]
== Inspect and download metrics

From the **Hosts** page, you can access a text-based view of the data underlying
your metrics visualizations and optionally download the data to a
comma-separated (CSV) file. 

Under **Metrics**, hover your cursor over a visualization, then in the
upper-right corner, click the Ellipses icon to inspect the data.

In the flyout, click **Download CSV** to download formatted or raw data to a CSV
file. 

Notice that you can change the view to **View: Requests** to explore the request
used to fetch the data and the response returned from {es}. You can click links
to further inspect and analyze the request in the **Dev Console** or
**Search Profiler**.

[discrete]
[[analyze-hosts-fields]]
== Field reference

The following sections show the formulas used to populate the **Hosts** page
with data.

****

> **Question for reviewers**

Author: @roshan-elastic This doc is not quite ready for review, but you could really help me out by a) indicating whether this info is useful, and b) provide a plain english description of how the value for each metric is calculated.

If you think providing the formula is good enough, then help me figure out the formula used by the tiles at the top of the page. I looked in the code, but it was taking too long so decided it's better to ask.

Reviewer: This is fantastic! Thank you!

Reviewer: Updated in 40b6ba4c1b4c613363b7e8b4b9106093c9c23cd3.

visualization to change the date range.
* In the visualizations under **Metrics**, click a point on a line and apply
filters to set other visualizations on the page to the same time and/or host.

To learn more about filtering data in {kib}, refer to
{kibana-ref}/kibana-concepts-analysts.html[{kib} concepts].

[NOTE]
.Limitation on filtering by service names
====
You may see queries where either the expected hosts are not returned, or where returned hosts do not return metrics.
This may happen when your filter returns events that do not contain all of the fields and/or values specified within your search.

For example, if you search for `service.name : myservice`, you will return all hosts where there are events
containing both `service.name : myservice` and where `host.name` exists.
You may find that some metrics show as `0` when there are no events which contain both
`host.name` and `service.name : myservice` and the metric you are looking for -- like `system.memory.actual.used.pct` for Memory Usage.

TIP: You can configure which indices you are searching in the Settings page.
====

[discrete]
[[analyze-hosts-open-in-lens]]
== Open in Lens

> Because all visualizations are powered by Lens, you can continue your analysis
> in Lens if you require more flexibility. Under **Metrics**, hover your cursor
> over a visualization, then click the ellipsis icon in the upper-right corner to
> open the visualization in Lens.

Reviewer: > Only the charts (at the bottom) are powered by lens. Do you think you could update this to infer that only the Metrics charts will open in lens?

Is "under Metrics" clear enough? Or do you prefer something like this:

```suggestion
Metrics visualizations are powered by Lens, meaning you can continue your analysis
in Lens if you require more flexibility. Under **Metrics**, hover your cursor
over a visualization, then click the ellipsis icon in the upper-right corner to
open the visualization in Lens.
```
